amqp/src/main/scala/akka/stream/alpakka/amqp/impl/AmqpSourceStage.scala

```diff
@@ -156,10 +160,12 @@ private[amqp] final class AmqpSourceStage(settings: AmqpSourceSettings, bufferSi
               pushMessage(queue.dequeue())
             }
 
-          override def onDownstreamFinish(): Unit = {
-            setKeepGoing(true)
-            if (unackedMessages == 0) super.onDownstreamFinish()
-          }
+          override def onDownstreamFinish(): Unit =
+            if (finishWithOutstandingAcks || unackedMessages == 0) super.onDownstreamFinish()
```

Comment:
I think that it's misleading to have a `unackedMessages` counter different to zero if messages are actually acknowledged.

It would be cleaner to do the check at line 168/174
(https://github.com/akka/alpakka/pull/1501/files#diff-79b48cb12e370d65fef5180dc16ab6aaR174)
like
```scala
      def pushMessage(message: CommittableIncomingMessage): Unit = {
        push(out, message)
        if (super.settings.ackRequired) unackedMessages += 1
      }
```

Reply:
You are right. 
Does it have any effect in case `ack` or `nack` are called when `autoAck` is `true`? 
Those could be short-circuited I guess. Or the source should not even emit committable messages...

Reply:
True.
Probably you should do the same at when the counter is decreased: `if (super.settings.ackRequired) unackedMessages -= 1` or even ``if (unackedMessages > 0) unackedMessages -= 1` to ensure that it doesn't go negative.

Reply:
I went a bit further and made ack/nack no-ops when using auto-ack.

Reply:
Looks great now.

However, I still think that the API is kind of broken (as discussed in #1167 ) since you can use the `withAckRequired` when creating a `committableSource` which doesn't make sense.

Reply:
Yes, you're right. I'll put an API might change there.

Reply:
I've created #1513

Reply:
Thanks!

Apart from that, the PR LGTM!
It fixes the actual issue and doesn't touch the current API.